add document on semantics options and add options to terminology document

pfps opened the pull request.

ktk committed:
add document on semantics options and add options to terminology document

gkellogg commented on docs/rdf-terminology.md:
```diff
@@ -134,3 +134,11 @@
+
+## New terms:
+
+* [Opaque semantics] – Any semantics for quoted triples where the syntactic form of the arguments to triples is important.  See [docs/semantics.md] for more information.
```
These don't work as Markdown. The form `[Opaque semantics]` would need a URL to reference, (e.g., `[Opaque semantics](/docs/semantics.md#opaque-semantics)`). Same for `[docs/semantics.md]`, which might be `[Semantics](/docs/semantics.md)`.

pfps replied:
@gkellogg I think I have something that works now.

TallTed commented on docs/rdf-terminology.md:
```diff
@@ -134,3 +134,11 @@
+
+* [Opaque semantics] – Any semantics for quoted triples where the syntactic form of the arguments to triples is important.  See [docs/semantics.md] for more information.
+
+* [Semi-opaque (semi-transparent semantics] – Any semantics for quoted triples where the syntactic form of the arguments to triples is important, except that only the meaning of blank nodes is important.  See [docs/semantics.md] for more information.
```
I know this punctuation change is appropriate. I'm also memorializing the need for rephrasing of "except that only the meaning of blank nodes is important" (which may also include rephrasing other parts of term description).
```suggestion
* [Semi-opaque (semi-transparent semantics)] – Any semantics for quoted triples where the syntactic form of the arguments to triples is important, _except that only the meaning of blank nodes is important_.  See [docs/semantics.md] for more information.
```

pfps replied:
@TallTed That phrase has been reworked to satisfy other concerns, but I did add the emphasis.

TallTed replied:
I only added the emphasis to highlight the phrase needing a rewrite... so I may suggest removing it when I see the rephrase.

pfps committed:
fix link syntax

TallTed commented on docs/semantics.md:
```diff
@@ -0,0 +1,87 @@
+
+The difference between the semantics can be seen in the following examples.
+
+In RDF (extended to include owl:sameAs as equality to provide a way of requiring that two different IRIs mean the same thing)
```
```suggestion
In RDF (extended to include `owl:sameAs` as equality to provide a way of requiring that two different IRIs mean the same thing)
```

rat10 commented:
> where the syntactic form of the arguments to triples is important, _except that only the meaning of blank nodes is important_.

As I read it the last (emphasized) half-sentence effectively overrules the preceding half-sentence. The meaning of the whole sentence IMO is that "only the meaning of blank nodes is important, everything else doesn't matter".
Also, this kind of language is still semanticists-speak. I would prefer a less concise, but more approachable version of "the meaning of blank nodes is important", e.g. "blank nodes are interpreted and mean the same as in neighbouring non-embedded triples, whereas other terms...". That "..." denotes a problem too - do syntactic terms in quoted triples refer to something in the realm of interpretation, just only in this specific syntactic form? If, as I assume, yes then that should be made more explicit as well.

TallTed commented on docs/semantics.md:
````diff
@@ -0,0 +1,87 @@
+
+In the opaque semantics
+```
+:b owl:sameAs :c .
+<<_:a :b "4"^^xsd:int >> :d :e .
````
Lines like 50-51 here exemplify why I want to see the optional whitespace at *both* ends of any given enclosure. `<<_:a` *looks* like it is one term, rather than being expandable to `<< _:a`.

Further, in documentation like this whole page, I find comprehensibility to be *vastly* increased by adding optional whitespace such that the columns visually align from start to finish, which would make these two lines look like the following:
```
:b                            owl:sameAs :c .
<< _:a :b "4"^^xsd:int     >> :d         :e .
```
— which match up with line 59, which has the widest quoted triple as well as the widest object within a quoted triple —
```
<< _:a :b "4"^^xsd:integer >> :d         :e .
```

pfps replied:
Yes, good idea.  I've fixed (to be uploaded soon).

TallTed commented on docs/rdf-terminology.md:
```diff
@@ -134,3 +134,11 @@
+
+* [Opaque semantics](/docs/semantics.md#opaque-semantics) – Any semantics for quoted triples where the syntactic form of the arguments to quoted triples is important.
+
+* [Semi-transparent semantics](/docs/semantics.md#semi-transparent-semantics) – Any semantics for quoted triples where the syntactic form of some, but not all, kinds of arguments to quoted triples is important _usually excepting blank nodes so that only their meaning is important_.
```
```suggestion
* [Semi-transparent semantics](/docs/semantics.md#semi-transparent-semantics) – Any semantics for quoted triples where the syntactic form of some, but not all, kinds of arguments to quoted triples is important _usually excepting blank nodes so that only their meaning (and not their syntactic form) is important_.
```

pfps replied:
As I already added the spaces, I'll make this change in my copy.

pchampin commented on docs/semantics.md:
```diff
@@ -0,0 +1,87 @@
+There have been several different semantics proposals for quoted triples, some with multiple semantics.  The proposed semantics differ in technical details but also differ in the entailments they support.  The major differences between semantics can be summarized in whether they are sensitive to the precise syntax of quoted triples or only to the semantics of the components of quoted triples.   T ...
+
+The difference between the semantics can be seen in the following examples.
+
```
```suggestion
Note that, although we use the Turtle syntax for convenience, we interpret it in a subtly different way: each blank node labels is interpreted  as specific and fixed element of the set of Blank Nodes defined by the RDF abstract syntax.

```
Without this comment, I would argue that it is impossible to decide whether the fisrt entailment holds or not in the opaque semantics (because we can't tell for sure if `_:a` and `_:f` are representing the same or different blank nodes...

afs replied:
Good to point it out. As defined in syntax currently, the scope of labels is the document. Same label, same blank node entity. Interpretation of the blank node may differ which is not a syntax issue.

pfps replied:
Agreed.  I've added this to my local copy and will upload shortly.

pchampin reviewed (commented):
Appart from the changes that I proposed, LGTM.

pfps committed:
Update docs/semantics.md

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pfps commented:
@rat10 The terminology bits are designed to be short, and they point to a longer document with more information.

pfps commented on docs/semantics.md:
```diff
@@ -0,0 +1,87 @@
+
+## Semi-transparent Semantics
+
+In the semi-transparent (a.k.a. semi-opaque) semantics
```
I added a full reference to the report

pfps committed:
incorporate suggestions

TallTed commented on docs/rdf-terminology.md:
```diff
@@ -134,3 +134,11 @@
+
+* [Opaque semantics](/docs/semantics.md#opaque-semantics) – Any semantics for quoted triples where the syntactic form of the arguments to quoted triples is important.
+
+* [Semi-transparent semantics](/docs/semantics.md#semi-transparent-semantics) – Any semantics for quoted triples where the syntactic form of some, but not all, kinds of arguments to quoted triples is important _usually excepting blank nodes so that only their meaning is (and not their syntactic form) important_.
```
```suggestion
* [Semi-transparent semantics](/docs/semantics.md#semi-transparent-semantics) – Any semantics for quoted triples where the syntactic form of some, but not all, kinds of arguments to quoted triples is important _usually excepting blank nodes so that only their meaning (and not their syntactic form) is important_.
```

TallTed commented on docs/semantics.md:
````diff
@@ -0,0 +1,89 @@
+
+In the opaque semantics
+```
+:b owl:sameAs :c .
+<< _:a :b "4"^^xsd:int >> :d :e .
+```
+does not entail
+```
+<< _:f :b "4"^^xsd:int >> :d :e .
+```
+or 
+```
+<< _:a :b "4"^^xsd:integer >> :d :e .
+```
+or
+```
+<< _:a :c "4"^^xsd:int >> :d :e .
+```
+
+## Semi-transparent Semantics
+
+In the semi-transparent (a.k.a. semi-opaque) semantics, as proposed in [the RDF-star final community group report](https://www.w3.org/2021/12/rdf-star.html)
+```
+:b owl:sameAs :c .
+<< _:a :b "4"^^xsd:int >> :d :e .
+```
+does entail
+```
+<< _:f :b "4"^^xsd:int >> :d :e .
+```
+but does not entail
+```
+<< _:a :b "4"^^xsd:integer >> :d :e .
+```
+or
+```
+<< _:a :c "4"^^xsd:int >> :d :e .
+```
+
+One could also create other semi-opaque semantics where either IRIs or literals were transparent.
````
Visually aligning the columns of triples, both the quoted and the quoting.

~~~suggestion
In RDF (extended to include `owl:sameAs` as equality to provide a way of requiring that two different IRIs mean the same thing)
```
:b                            owl:sameAs :c .
_:a                           :b         "4"^^xsd:int .
```
entails
```
_:f                           :b         "4"^^xsd:int .
```
and 
```
_:a                           :b         "4"^^xsd:integer .
```
and 
```
_:a                           :c         "4"^^xsd:integer .
```


## Transparent Semantics

In the _transparent_ semantics
```
:b                            owl:sameAs :c .
<< _:a :b "4"^^xsd:int     >> :d         :e .
```
entails
```
<< _:f :b "4"^^xsd:int     >> :d         :e .
```
and 
```
<< _:a :b "4"^^xsd:integer >> :d         :e .
```
and
```
<< _:a :c "4"^^xsd:int     >> :d         :e .
```

## Opaque Semantics

In the _opaque_ semantics
```
:b owl:sameAs :c .
<< _:a :b "4"^^xsd:int     >> :d         :e .
```
does not entail
```
<< _:f :b "4"^^xsd:int     >> :d         :e .
```
or 
```
<< _:a :b "4"^^xsd:integer >> :d         :e .
```
or
```
<< _:a :c "4"^^xsd:int     >> :d         :e .
```

## Semi-transparent Semantics

In the _semi-transparent_ (also known as _semi-opaque_) semantics, as proposed in [the RDF-star final community group report](https://www.w3.org/2021/12/rdf-star.html)
```
:b owl:sameAs :c .
<< _:a :b "4"^^xsd:int     >> :d         :e .
```
does entail
```
<< _:f :b "4"^^xsd:int     >> :d         :e .
```
but does not entail
```
<< _:a :b "4"^^xsd:integer >> :d         :e .
```
or
```
<< _:a :c "4"^^xsd:int     >> :d         :e .
```

One could also create other semi-opaque semantics where either IRIs or literals were transparent.
~~~

pfps committed:
Update docs/semantics.md

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

pfps committed:
Update docs/rdf-terminology.md

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

rat10 committed:
Update semantics.md

Keep the use of xsd:int for the forth variant consistent across different examples

rat10 committed:
Update semantics.md to improve example understanding

In my own and painful experience some small hints can make it much easier for non-logicians to understand what is going on and what an example tries to convey. 
The concrete changes I propose may not be fully correct or grasp what's going to the fullest extend, so take them as a prototype to illustrate the issue.
I deliberately used the term 'means' instead of 'denotes' or 'refers to' beacuse it is more approachable, but the latter might be more suitable as they are used much more frequently in the specs.

rat10 committed:
Update semantics.md to better describe the meaning of opacity

The document should explain in greater detail to non-logicians (which I assume to be its main target audience, right?) what it means that the syntax of a term is important. It doesn't mean that the term is just a string (right?!) but that its interpretation is restricted. This pull request tries to explain it, and although it probably fails some explanation IMO is necessary.

pfps committed:
Merge pull request #69 from rat10/patch-1

Update semantics.md to improve consistency across example sections

pfps committed:
Merge pull request #73 from rat10/patch-5

Update semantics.md to better describe the meaning of opacity

rat10 committed:
Update semantics.md

fixed typo

pfps committed:
Merge pull request #72 from rat10/patch-4

Update semantics.md to improve example understanding
